### ISSUE TITLE
TPLs (Third Party Libraries) for lfric-jedi [optional]

### ISSUE THREAD

**Issue by yaswant** [JCSDA/spack-stack#737]
**Please describe the package or library you would like to add to spack-stack.**

This is a _placeholder_ to add third-party libraries required to compile `lfric-jedi` applications. 

A description of packages including GitHub repo.

- bison  https://ftp.gnu.org/gnu/bison
- blitz  https://github.com/blitzpp/blitz
- yaxt  https://swprojects.dkrz.de/redmine/projects/yaxt/wiki/Downloads
- xios  http://forge.ipsl.jussieu.fr/ioserver/svn/XIOS
- Psyclone  https://github.com/stfc/PSyclone
- ~~lfricapi-utils~~ rose_picker https://github.com/MetOffice/rose_picker
- ~~lfricapi~~ lfric_core https://github.com/MetOffice/lfric_core
- lfric_apps https://github.com/MetOffice/lfric_apps
- shumlib https://github.com/MetOffice/shumlib
    
**What applications will be using this package or library?**
- lfric-jedi

**Is there already a package or library in spack-stack that provides this, or related, functionality?**
NA

**Additional context**
NA

**Will This Package be Needed in a NOAA Operational Application?**
No.  This is required for Met Office specific applications only.

**Comment by climbfuji** [comment]
@yaswant Can you help me understand the need for `shumlib`? Is this something that was needed for the UM only, not for `lfric`/`lfric-lite`? If so, when can we remove it from spack-stack? Thanks!

**Comment by yaswant** [comment]
@climbfuji thats right. However ops-um-jedi is not redundant just yet. Is shumlib proving difficult in your stack? If not, then please can we keep it for the moment? We use the Jedi containers here from time to time to test/compare.

**Comment by climbfuji** [comment]
It's not difficult to keep. We are using a branch that sits in my fork, because the original shumlib commit had serious flaws that prevented building it on macOS and with most compilers except the one used by default (gnu?). We were never able to merge those changes back. But if the lifetime of this packages is limited to another year or so, we can continue doing what we did for the last two years and use my fork/branch.

**Comment by yaswant** [comment]
Hi @climbfuji - have you looked at

**Comment by yaswant** [comment]
Thanks @climbfuji . Looking at https://github.com/metomi/shumlib/pull/1 have you tested the latest changes on Mac?  We are currently using `um13.0` tag in our stack, could you please try that and update your shumlib PR?  I'll then remind @jennyhickson to merge any changes if required.

**Comment by eap** [comment]
@yaswant - what's the status of this, have you worked around this issue or is this still blocking work? We're going through old issues and seeing if they need to be added to the roadmap.

**Comment by yaswant** [comment]
We have recently update public facing shumlib - thanks to Matt S. can you please try https://github.com/metomi/shumlib/releases/tag/um13.9 and see if the issue is resolved?

**Comment by climbfuji** [comment]
Is JCSDA going to work on this?

Related, is `shumlib` still needed or can we remove it?

**Comment by eap** [comment]
@yaswant - this issue is still difficult for me to understand. Can you help me understand exactly the desired end-state here? Which of these libraries are still needed, what versions, and what environments are they needed in? For packages lfricapi-utils and lfricapi, where are these published?

I'll summarize the current state to be best of my knowledge.

1) The original request was for bison, blitz, yaxt, xios, Psyclone, lfricapi-utils, and lfricapi. Here's the current status of each of these projects

| Package | spack-stack status | spack-packages availability| public availability | 
| ---------|-------------------|-----------------|---------|
| bison | CI Containers, jedi-base-env, various sites | yes; v2.7-v3.82 | yes |
| blitz  | not installed | yes; v1.0.2 | yes |
| xios | not installed | yes; v2.6 | yes |
| psyclone | not installed | yes; v1.5->3.1 | yes |
| yaxt | not installed | no | yes |
|  lfricapi-utils | not installed | no | no? |
| lfricapi | not installed | no | no? |

2) Secondary to the original request, there was a request for shumlib. This is currently served by a [spack-ext](https://github.com/JCSDA/spack-stack/blob/develop/spack-ext/repos/spack-stack/packages/shumlib/package.py) package in spack-stack. Is the version referenced there still adequate? Is the package still needed?

**Comment by climbfuji** [comment]
I'll add that for shumlib, we are still pointing to a branch in my fork, because the changes that were needed for apple-clang (and Intel I believe) never made it into the authoritative code. But maybe `shumlib` is no longer needed with lfric?

**Comment by matthewrmshin** [comment]
Update following quick chat with @yaswant:

- lfricapi-utils => rose_picker - this is a little Python package that will become publicly available.
- lfricapi => lfric (core) and lfric_apps - both will become publicly available soon.

**Comment by yaswant** [comment]
We have [rose_picker](/MetOffice/rose_picker), [lfric_core](/MetOffice/lfric_core), [lfric_apps](/MetOffice/lfric_apps), and more recent version of [shumlib](/MetOffice/shumlib) in Met Office public domain now.

**Comment by climbfuji** [comment]
@eap @yaswant Can you please follow up on this?
